nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java

```diff
@@ -0,0 +1,229 @@
+    }
+
+    @Test
+    public void shouldFailWhenRetrievingDocumentBlobWithoutColdStorageContent() {
+        DocumentModel documentModel = createDocument(DEFAULT_DOC_NAME, true);
+        try {
+            // try to retrieve from cold storage where the blob is not stored in it
+            ColdStorageHelper.retrieveContentFromColdStorage(session, documentModel.getRef(),
+                    NUMBER_OF_DAYS_OF_AVAILABILITY);
+            fail("Should fail because there no cold storage content associated to this document.");
+        } catch (NuxeoException ne) {
+            assertEquals(SC_NOT_FOUND, ne.getStatusCode());
+            assertEquals(String.format("No cold storage content defined for document: %s.", documentModel),
+                    ne.getMessage());
+        }
+    }
+
+    @Test
+    public void shouldCheckAvailabilityOfColdStorageContent() {
+        List<String> documents = Arrays.asList( //
+                moveAndRetrieveColdStorageContent(DEFAULT_DOC_NAME).getId(),
+                moveAndRetrieveColdStorageContent("anyFile2").getId(), //
+                moveAndRetrieveColdStorageContent("anyFile3").getId());
+        transactionalFeature.nextTransaction();
+        try (CapturingEventListener listener = new CapturingEventListener(
+                ColdStorageHelper.COLD_STORAGE_CONTENT_AVAILABLE_EVENT_NAME)) {
+            ColdStorageHelper.ColdStorageContentStatus coldStorageContentStatus = ColdStorageHelper.checkColdStorageContentAvailability(
+                    session);
+            assertEquals(session.getRepositoryName(), coldStorageContentStatus.getRepositoryName());
+
+            // all cold storage contents being retrieved are now available
+            assertEquals(0, coldStorageContentStatus.getTotalBeingRetrieved());
+            assertEquals(documents.size(), coldStorageContentStatus.getTotalContentAvailable());
+
+            assertTrue(listener.hasBeenFired(ColdStorageHelper.COLD_STORAGE_CONTENT_AVAILABLE_EVENT_NAME));
+            assertEquals(3, listener.streamCapturedEvents().count());
+
+            List<String> docEventIds = listener.streamCapturedEvents() //
+                                               .map(event -> {
+                                                   DocumentEventContext docCtx = (DocumentEventContext) event.getContext();
+                                                   return docCtx.getSourceDocument().getId();
+                                               }) //
+                                               .sorted() //
+                                               .collect(Collectors.toList());
+
+            documents.sort(Comparator.naturalOrder());
+            assertEquals(documents, docEventIds);
+        }
+    }
+
+    protected DocumentModel moveAndRetrieveColdStorageContent(String documentName) {
+        DocumentModel documentModel = createDocument(documentName, true);
+        ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
```

Comment:
Here also I would put `documentModel = ColdStorageHelper.moveContentToColdStorage`, just to always use the same pattern.